.github/workflows/deploy-changed-cf.yaml

```diff
@@ -88,29 +88,105 @@ jobs:
         name: Set name of the CloudFormation stack
         run: |-
           echo "Ref is $GITHUB_REF"
-          BASE_NAME=biomage-$(basename $FILE_NAME | sed "s/\..*//")-$CLUSTER_ENV
-          echo "::set-output name=stack-name::$BASE_NAME"
+          BASE_NAME=$(basename $FILE_NAME | sed "s/\..*//")
+          STACK_NAME=biomage-$BASE_NAME-$CLUSTER_ENV
+
+          echo "::set-output name=file-name::$BASE_NAME"
+          echo "::set-output name=stack-name::$STACK_NAME"
         env:
           FILE_NAME: ${{ matrix.template }}
 
-      - id: deploy-template
-        name: Deploy CloudFormation template
-        if: ${{ !contains(matrix.template, 'irsa-') && github.ref == 'refs/heads/master' }}
+      - id: deploy-irsa-template
+        if: ${{ contains(matrix.template, 'irsa-') && github.ref == 'refs/heads/master' }}
+        name: Deploy IRSA CloudFormation template
         uses: aws-actions/aws-cloudformation-github-deploy@v1
         with:
-          parameter-overrides: "Environment=${{ matrix.environment }}"
+          parameter-overrides: "Environment=${{ matrix.environment }},OIDCProvider=${{ steps.get-oidc.outputs.oidc-provider }}"
           name: ${{ steps.set-name.outputs.stack-name }}
           template: ${{ matrix.template }}
           no-fail-on-empty-changeset: "1"
           capabilities: "CAPABILITY_IAM,CAPABILITY_NAMED_IAM"
 
-      - id: deploy-irsa-template
-        if: ${{ contains(matrix.template, 'irsa-') && github.ref == 'refs/heads/master' }}
-        name: Deploy IRSA CloudFormation template
+      - id: deploy-template
+        if: ${{ !contains(matrix.template, 'irsa-') && github.ref == 'refs/heads/master' }}
         uses: aws-actions/aws-cloudformation-github-deploy@v1
+        name: Deploy CloudFormation template
         with:
-          parameter-overrides: "Environment=${{ matrix.environment }},OIDCProvider=${{ steps.get-oidc.outputs.oidc-provider }}"
```

Comment:
Why was OIDCProvider removed?

Reply:
I'm switching the position of the `non-irsa-` (does not require OIDC) and `irsa-` (require OIDC) resources